### PR TITLE
Give all Items from via Lua

### PR DESCRIPTION
**Pull request by ThanatosGit**
It shows the "OK" button at the message box directly but don't know how to change that.

### CODE REVIEW

File: open_samus_returns_rando/files/randomizer_powerup.lua [hunk 1]
```diff
@@ -139,192 +126,161 @@ function RandomizerPowerup.IncreaseAmmo(resource)
...
-    if main_item then
-        new_current = current_max + locked_pbs
-    else
-        new_current = current_current + locked_pbs
-    end
-    RandomizerPowerup.SetItemAmount("ITEM_WEAPON_POWER_BOMB_MAX", current_max + locked_pbs)
-    RandomizerPowerup.SetItemAmount("ITEM_WEAPON_POWER_BOMB_CURRENT", new_current)
-    RandomizerPowerup.SetItemAmount("ITEM_RANDO_LOCKED_PBS", 0)
-    hud.UpdatePlayerInfo(true)
+-- TODO: What's the max energy?
```
Comment (ThanatosGit) [inline open_samus_returns_rando/files/randomizer_powerup.lua:129]: ?

File: open_samus_returns_rando/files/randomizer_powerup.lua [hunk 2]
```diff
@@ -139,192 +126,161 @@ function RandomizerPowerup.IncreaseAmmo(resource)
...
+    new_max = math.min(new_max, MAX_ENERGY)
+    RandomizerPowerup.SetItemAmount("ITEM_MAX_LIFE", new_max)
+    RandomizerPowerup.SetItemAmount("ITEM_CURRENT_LIFE", new_max)
+
+    local life = Game.GetPlayer().LIFE
+    life.fMaxLife = new_max
+    life.fCurrentLife = new_max
+end
+
+-- TODO: What's the max aeion?
```
Comment (ThanatosGit) [inline open_samus_returns_rando/files/randomizer_powerup.lua:143]: ?

File: open_samus_returns_rando/files/randomizer_powerup.lua (hunk 1 above)
Comment (dyceron) [reply open_samus_returns_rando/files/randomizer_powerup.lua:129]: 1099

File: open_samus_returns_rando/files/randomizer_powerup.lua (hunk 2 above)
Comment (dyceron) [reply open_samus_returns_rando/files/randomizer_powerup.lua:143]: 2200

**Comment by dyceron** [comment]
I assume the max values for energy and aeion aren't hard caps, because I'm able to set my max to be much higher than these.

File: open_samus_returns_rando/files/randomizer_powerup.lua (hunk 1 above)
Comment (ThanatosGit) [reply open_samus_returns_rando/files/randomizer_powerup.lua:129]: Does the game do not explode at refill stations? Iirc that was the issue with dread: You could go higher but have crashes at refill stations.

Comment (dyceron) [reply open_samus_returns_rando/files/randomizer_powerup.lua:129]: I don't recall having issues

**Comment by ThanatosGit** [comment]
Thanks for the fixes.
`ITEM_CURREITEM_CURRENT_SPECIAL_ENERGYNT_LIFE` that happens way too often for me.
Copy paste something from the left to the right, but miss the double click on the right when selecting the string to replace. 😔

**Review by dyceron — APPROVED** [review]

**Pull request merged**